server/test/auth.test.js

```diff
@@ -220,4 +220,113 @@ describe('Auth Routes', () => {
         done();
       });
   });
+
+
+  it('should log in an existing user', (done) => {
+    const user = {
+      email: 'ryan@gmail.com',
+      password: 'ryangosl'
+    };
+    request(app)
+      .post(`${URL}/login`)
+      .send(user)
+      .expect(200)
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body).to.have.property('status').equal(200);
+        expect(res.body).to.have.property('message').equal('Login successful');
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  it('should not log in a user with empty email and password fields', (done) => {
+    const user = {
+      email: '',
+      password: ''
+    };
+    request(app)
+      .post(`${URL}/login`)
+      .send(user)
+      .expect(400)
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        expect(res.body).to.have.property('status').equal(400);
+        expect(res.body).to.have.property('error').equal('Email and password is required');
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  it('should not log in a user with an empty email field', (done) => {
+    const user = {
+      email: '',
+      password: 'ryangosl'
+    };
+    request(app)
+      .post(`${URL}/login`)
+      .send(user)
+      .expect(400)
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        expect(res.body).to.have.property('status').equal(400);
+        expect(res.body).to.have.property('error').equal('Email is required');
+        if (err) return done(err);
+        done();
+      });
+  });
+
+  it('should not log in a user with an empty password field', (done) => {
+    const user = {
+      email: 'ryan@gmail.com',
+      password: ''
+    };
+    request(app)
+      .post(`${URL}/login`)
+      .send(user)
+      .expect(400)
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        expect(res.body).to.have.property('status').equal(400);
+        expect(res.body).to.have.property('error').equal('Password is required');
```

Comment:
Great, great job. Please remember to add a test for bad emails when you implement the validation

Reply:
done..